src/classes/CON_ContactMerge_CTRL.cls

```diff
@@ -274,6 +274,14 @@ public with sharing class CON_ContactMerge_CTRL {
             Schema.SobjectField f = contactFields.get(fieldName);
             Schema.DescribeFieldResult fResult = f.getDescribe();
             
+            // Ignore fields that have an identical API name as an npsp prefixed field
+            // This is due to a bug in dynamic SOQL, quich causes querying for two real fields, 
+            // e.g. npsp__foo__c and foo__c, to throw a duplicate field selected error
+            if (fieldName != UTIL_Namespace.StrTokenRemoveNSPrefix(fieldName) && 
+                contactFields.containsKey(UTIL_Namespace.StrTokenRemoveNSPrefix(fieldName))) {
+                continue;
+            }
+
```

Comment:
My understanding is that dynamic SOQL resolves both fields, whether with or without namespace, to the NPSP namespaced field, which means no matter what we as package developers do, we can only show our version of the field in our UI. I could definitely update the logic to make that more clear.